bananas_api/tool_region/__main__.py

```diff
@@ -0,0 +1,113 @@
+def main(m49_csv_file, iso_3166_2_json_file):
+    regions = set()
+
+    with open(m49_csv_file) as fp:
+        for line in fp.readlines()[1:]:
+            line = line.strip().split(";")
+
+            if len(line) != 15:
+                raise Exception("Invalid line; is this an UN M49 CSV file?")
+
+            (
+                global_code,
+                _,
+                region_code,
+                region_name,
+                sub_region_code,
+                sub_region_name,
+                intermediate_region_code,
+                intermediate_region_name,
+                country,
+                _,
+                country_code,
+                _,
+                _,
+                _,
+                _,
+            ) = line
+
+            if global_code != "001":
+                raise Exception("Invalid global code; is this an UN M49 CSV file from earth?")
+
+            if not country_code:
+                print(f"INFO: skipping '{country}' as it has no ISO-3361-1 country code")
+                continue
+
+            # If the intermediate region is set, ignore the sub region.
+            # This is because in these cases the sub region tends to be a long
+            # name and doesn't add information to the (intermediate) region.
+            if intermediate_region_code:
+                sub_region_code = intermediate_region_code
+                sub_region_name = intermediate_region_name
+
+            # Prefix UN specific codes with UN-. This to give them a visual
+            # difference from the ISO codes.
+            if region_code:
+                region_code = f"UN-{region_code}"
+            if sub_region_code:
+                sub_region_code = f"UN-{sub_region_code}"
+
+            regions.add((country_code, country, sub_region_code))
+            if sub_region_code:
+                regions.add((sub_region_code, sub_region_name, region_code))
+            if region_code:
+                regions.add((region_code, region_name, ""))
+
+    with open(iso_3166_2_json_file) as fp:
+        data = json.load(fp)
+
+        for country in data["3166-2"]:
+            subdivision_code = country["code"]
+            # Normalize all names to be within ASCII. This makes searching in-game easier.
+            subdivision_name = unicodedata.normalize("NFKD", country["name"]).encode("ascii", "ignore").decode()
+
+            # There are several ways to denote aliases; strip those out.
+            subdivision_name = subdivision_name.lstrip("/")
+            subdivision_name = subdivision_name.split("/")[0].strip()
+            subdivision_name = subdivision_name.split("(")[0].strip()
+            subdivision_name = subdivision_name.split("[")[0].strip()
+
+            country_code = subdivision_code[:2]
+
+            regions.add((subdivision_code, subdivision_name, country_code))
+
+    cwd = os.path.dirname(__file__)
+
+    with open(f"{cwd}/../helpers/regions.py", "w") as fp:
```

Comment:
I'm never a fan of scripts that decide where to output things for me. Perhaps stdout instead, with stderr for the INFO message?

Being able to just use `print()` would simplify having to do all those `\n`...

Reply:
This is a lot of output, and the tool is designed to bring this to a very specific file. Important to observe it doesn't use your current working directory, but the location of the source file to find the correct location.

So I am not sure why you would prefer output to stdout, which means you have to pipe it yourself to this file? Feels like more work with no added benefit? Asking, as I might be missing something :)

Reply:
Just personal preference. As the advantage of being able to preview output (or even diff it) before letting it trample over my working dir

Reply:
The file is in git, so you can do all that! And that without accidentally getting 5000 lines of code over your stdout :)

Reply:
After some fiddling around, this solution just sucks in general; and this is just a symptom.

I am going to add the CSV and JSON file to the repository, and load them on start-up. That way it is much more clear what is going on, etc.